sdk/storage/Azure.Storage.Common/src/Shared/WriteStream.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Storage.Shared
+{
+    internal abstract class WriteStream : Stream
+    {
+        protected long _position;
+        protected readonly IProgress<long> _progressHandler;
+        protected readonly MemoryStream _buffer;
```

Comment:
This isn't going to work for jumbo blobs. I think users of jumbo blobs/blocks are going to be part of audience for this feature. I think we should borrow solution from partitioned uploader to handle that here.

Reply:
Switched to PooledMemoryStream.